commcare-cloud/commcare_cloud/environment/schemas/postgresql.py

```diff
@@ -1,39 +1,108 @@
 import jsonobject
+import re
+
+import six
+
+from commcare_cloud.environment.constants import constants
+
+
+def alphanum_key(key):
+    """
+    Used to generate a natural numeric sort key
+
+    Example: ("p10" => ["p", 10]) is greater than ("p1" => ["p", 1])
+
+    adapted from https://stackoverflow.com/a/2669120/240553
+    """
+    def convert(text):
+        return int(text) if text.isdigit() else text
+    return [convert(c) for c in re.split(r'([0-9]+)', key)]
 
 
 class PostgresqlConfig(jsonobject.JsonObject):
     _allow_dynamic_properties = False
 
     SEPARATE_SYNCLOGS_DB = jsonobject.BooleanProperty(default=True)
+    SEPARATE_FORM_PROCESSING_DBS = jsonobject.BooleanProperty(default=True)
     DEFAULT_POSTGRESQL_HOST = jsonobject.StringProperty(default=None)
     DEFAULT_POSTGRESQL_USER = jsonobject.StringProperty(default="{{ secrets.POSTGRES_USERS.commcare.username }}")
     DEFAULT_POSTGRESQL_PASSWORD = jsonobject.StringProperty(default="{{ secrets.POSTGRES_USERS.commcare.password }}")
-
-    postgresql_dbs = jsonobject.ListProperty(lambda: DBOptions, required=True)
+    REPORTING_DATABASES = jsonobject.DictProperty(default=lambda: {"ucr": "ucr"})
+    dbs = jsonobject.ObjectProperty(lambda: SmartDBConfig)
 
     @classmethod
     def wrap(cls, data):
         self = super(PostgresqlConfig, cls).wrap(data)
-        for db in self.postgresql_dbs:
+        for db in self.generate_postgresql_dbs():
             if not db.user:
                 db.user = self.DEFAULT_POSTGRESQL_USER
             if not db.password:
                 db.password = self.DEFAULT_POSTGRESQL_PASSWORD
         return self
 
+    def to_json(self):
+        data = super(PostgresqlConfig, self).to_json()
+        data['postgresql_dbs'] = [db.to_json() for db in self.generate_postgresql_dbs()]
+        return data
+
     def replace_hosts(self, environment):
         if self.DEFAULT_POSTGRESQL_HOST is None:
             self.DEFAULT_POSTGRESQL_HOST = environment.groups['postgresql'][0]
         elif self.DEFAULT_POSTGRESQL_HOST != '127.0.0.1':
             self.DEFAULT_POSTGRESQL_HOST = environment.translate_host(
                 self.DEFAULT_POSTGRESQL_HOST, environment.paths.postgresql_yml)
 
-        for db in self.postgresql_dbs:
+        for db in self.generate_postgresql_dbs():
             if db.host is None:
                 db.host = self.DEFAULT_POSTGRESQL_HOST
             elif db.host != '127.0.0.1':
                 db.host = environment.translate_host(db.host, environment.paths.postgresql_yml)
 
+    def generate_postgresql_dbs(self):
+        return filter(None, [
+            self.dbs.main, self.dbs.synclogs,
+        ] + (
+            self.dbs.form_processing.get_db_list() if self.dbs.form_processing else [None]
```

Comment:
nit: why `[None]` instead of `[]`?

Reply:
No good reason. I was thinking of `None` as the value of `self.dbs.form_processing`, but agree your suggestion is better. I'll change to `[]`.